holidays/countries/bermuda.py

```diff
@@ -0,0 +1,141 @@
+    ObservedHolidayBase,
+    SAT_SUN_TO_NEXT_MON,
+    SAT_SUN_TO_NEXT_MON_TUE,
+)
+
+
+class Bermuda(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, StaticHolidays):
+    """Bermuda holidays.
+
+    References:
+        * [Public Holidays Act 1947](https://web.archive.org/web/20250527163956/https://www.bermudalaws.bm/Laws/Consolidated%20Law/1947/Public%20Holidays%20Act%201947)
+        * [Public Holidays Amendment Act 1999](https://web.archive.org/web/20250527163749/https://www.bermudalaws.bm/Laws/Annual%20Law/Acts/1999/Public%20Holidays%20Amendment%20Act%201999)
+        * [Public Holidays Amendment Act 2008](https://web.archive.org/web/20250527163810/https://www.bermudalaws.bm/Laws/Annual%20Law/Acts/2008/Public%20Holidays%20Amendment%20Act%202008)
+        * [Public Holidays Amendment (No. 2) Act 2009](https://web.archive.org/web/20250527163816/https://www.bermudalaws.bm/Laws/Annual%20Law/Acts/2009/Public%20Holidays%20Amendment%20%28No.%202%29%20Act%202009)
+        * [Public Holidays Amendment Act 2017](https://web.archive.org/web/20250527163819/https://www.bermudalaws.bm/Laws/Annual%20Law/Acts/2017/Public%20Holidays%20Amendment%20Act%202017)
+        * [Public Holidays Amendment Act 2020](https://web.archive.org/web/20250527163836/https://www.bermudalaws.bm/Laws/Annual%20Law/Acts/2020/Public%20Holidays%20Amendment%20Act%202020)
+        * [Government of Bermuda](https://web.archive.org/web/20250530055854/https://www.gov.bm/public-holidays)
+        * [Proclamation BR 149 / 2018](https://web.archive.org/web/20250530114331/https://www.bermudalaws.bm/Laws/Annual%20Law/Statutory%20Instruments/2018/Proclamation%20-%20Bermuda%20Day%20Public%20Holiday)
+    """
+
+    country = "BM"
+    observed_label = "%s (observed)"
+    start_year = 1948
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        StaticHolidays.__init__(self, BermudaStaticHolidays)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day("New Year's Day"))
+
+        # Good Friday.
+        self._add_good_friday("Good Friday")
+
+        # Bermuda Day.
+        name = "Bermuda Day"
+        if self._year <= 2017:
+            self._add_observed(self._add_holiday_may_24(name))
+        elif self._year == 2019:
+            self._add_holiday_may_24(name)
+        elif self._year <= 2020:
+            self._add_holiday_last_fri_of_may(name)
+        else:
+            self._add_holiday_3_days_prior_last_mon_of_may(name)
+
+        # Queen's Birthday.
+        name = "Queen's Birthday"
+        if self._year <= 1999:
+            self._add_holiday_3_days_prior_last_mon_of_may(name)
+        elif self._year <= 2008:
+            self._add_holiday_2_days_past_2nd_mon_of_jun(name)
```

Comment:
```suggestion
            self._add_holiday_3rd_mon_of_june(name)
        elif self._year <= 2008:
            self._add_holiday_2_days_past_2nd_sat_of_jun(name)
```
Public Holidays Amendment Act 1999:
> in the entry relating to the Queen's Birthday by deleting the words "the third Monday in June" and substituting the words "the Monday following the second Saturday in June";